### PR TITLE
Add resource tagging support

### LINKED ISSUE #598: Allow setting instance tags for cloud providers
One thing I've noticed after looking through the docs here: https://registry.terraform.io/providers/iterative/iterative/latest/docs/resources/task#argument-reference

is that there is no way to assign a tag to an ec2 instance. This is often necessary for billing purposes, so it would be great if there were a way do this.

Thanks for the great work!

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
```hcl
resource "iterative_task" "example" {
  ···
  tags = {
    Something = "Useful"
  }
}
```

Closes #598

### CODE REVIEW

File: docs/resources/task.md [hunk 1]
```diff
@@ -64,6 +64,7 @@ resource "iterative_task" "example" {
 - `storage.workdir` - (Optional) Local working directory to upload and use as the `script` working directory.
 - `storage.output` - (Optional) Results directory (**relative to `workdir`**) to download (default: no download).
 - `environment` - (Optional) Map of environment variable names and values for the task script. Empty string values are replaced with local environment values. Empty values may also be combined with a [glob](<https://en.wikipedia.org/wiki/Glob_(programming)>) name to import all matching variables.
+- `tags` - (Optional) Map of tags for the created cloud resources.
 - `timeout` - (Optional) Maximum number of seconds to run before instances are force-terminated. The countdown is reset each time TPI auto-respawns a spot instance.
```
Comment (casperdcl) [inline docs/resources/task.md:68]: lower priority IMO :nail_care: :)

```suggestion
- `timeout` - (Optional) Maximum number of seconds to run before instances are force-terminated. The countdown is reset each time TPI auto-respawns a spot instance.
- `tags` - (Optional) Map of tags for the created cloud resources.
```

**Review by casperdcl — CHANGES_REQUESTED** [review]
lgtm

**Comment by dacbd** [comment]
I go to bed and he beats me to it 😆

**Comment by 0x2b3bfa0** [comment]
> _he beats me to it_

Sleeping, or opening pull requests? 🦦

**Review by casperdcl — APPROVED** [review]

**Comment by dacbd** [comment]
I might be being extra picky but for AWS as the example, the auto-scaling group doesn't get the tag, and more importantly, s3 objects are missing them as well.

**Comment by 0x2b3bfa0** [comment]
You're picky and it's awesome! 🙏🏼 Tracking this on https://github.com/iterative/terraform-provider-iterative/issues/601

**Pull request merged**